hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -450,60 +450,20 @@ private void validateLedgerState() {
 
     /*==================================================================================================================
     *
-    * Run the app.
+    * Other app lifecycle methods
     *
     =================================================================================================================*/
 
     /**
      * {@inheritDoc}
-     * <p>
-     * Called by the platform after <b>ALL</b> initialization to start the gRPC servers and begin operation.
+     *
+     * <p>Called by the platform after <b>ALL</b> initialization to start the gRPC servers and begin operation, or by
```

Comment:
At least from the platform's perspective, there are no restrictions as to when it is legal to call `boolean Platform.submitTransaction()`. This method may return false to signal that the transaction is rejected for whatever reason (perhaps the queue is full, or we are in the wrong lifecycle phase). If this method returns `true` then the transaction is scheduled to be submitted whenever it is legal for the platform to do so (although this is a "best effort" thing, the transaction is not guaranteed to eventually reach consensus or to reach consensus in any particular time span).

Reply:
The `run()` method on this class is called after the platform has been started. Note that this could mean that the platform may call various notifications concurrently with this method.

Reply:
```
        platforms.forEach(SwirldsPlatform::start);         // Platform starts doing stuff on many threads

        for (int nodeIndex = 0; nodeIndex < appRunThreads.length; nodeIndex++) {
            appRunThreads[nodeIndex].start();             // this calls run() on a background thread
        }
```

Reply:
OK, I have refactored these methods, such that the grpc server manager is non-null, always, and it's start/stop methods are synchronized. That should take care of the concurrent access.